### PR TITLE
[WIP] Add Calico to kube-up

### PR DESCRIPTION
**Pull request by caseydavenport**

### CODE REVIEW

**Comment by tnine** [comment]
Looks good so far.  I made 1 change in the "calico" branch.

https://github.com/30x/docker-compose-kubernetes/commit/57fdb2e50231c06c69edca99e5364f27e3c32cb0

That way your docker-machine name can be anything and it works.

**Comment by caseydavenport** [comment]
Looks good.

There is still an ordering concern:

the `calico/ctl` container in the compose file relies on `etcd` being available, but this is not always the case at the time compose tries to run that container.

Similar to this issue: https://github.com/docker/compose/issues/374

File: kubernetes/calico-node.env [hunk 1]
```diff
@@ -0,0 +1,6 @@
+ETCD_AUTHORITY=127.0.0.1:4001
+HOSTNAME=dev
+IP="10.0.2.15"
```
Comment (tnine) [inline kubernetes/calico-node.env]: This doesn't exist for me.  Should this be the docker host?

IP=$(echo $DOCKER_HOST | awk -F'[/:]' '{print $4}')

Comment (caseydavenport) [inline kubernetes/calico-node.env]: Yeah that is a better solution.

This was another area I wanted to clean up since it is likely tied to my local setup.

**Comment by jbowen93** [comment]
Something is making the kube-ui endpoint throw back a 503 error. I'm gonna try to debug the issue and if I get it sorted I'll merge the pull request.

**Pull request closed without merging**